crates/bevy_tasks/src/task_pool.rs

```diff
@@ -121,13 +121,23 @@ impl TaskPool {
                 let ex = Arc::clone(&executor);
                 let shutdown_rx = shutdown_rx.clone();
 
+                #[cfg(not(miri))]
```

Comment:
It feels like we could reduce the complexity here a bit by consolidating under a single pair of `#[cfg(miri)]` and `#[cfg(not(miri))]` flags